synapse/storage/registration.py

```diff
@@ -144,3 +144,21 @@ def _query_for_auth(self, txn, token):
             return rows[0]
 
         raise StoreError(404, "Token not found.")
+
+    @defer.inlineCallbacks
+    def all_users_on_domain(self, domain):
```

Comment:
The name made me think it was a way to list all the users on a domain. I guess I'd prefer it to be "are_all_users_on_domain" and to have some docs to say what the method does.